.github/workflows/validate-config-extractor.yaml

```diff
@@ -0,0 +1,79 @@
+# (C) Copyright Confidential Containers Contributors
+# SPDX-License-Identifier: Apache-2.0
+#
+# Validate config-extractor tool
+---
+name: validate-config-extractor
+
+on:
+  pull_request:
+    paths:
+      - 'src/cloud-providers/**/manager.go'
+      - 'src/cloud-providers/util.go'
+      - 'src/cloud-providers/cmd/config-extractor/**'
+  workflow_dispatch:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+permissions: {}
+
+jobs:
+  test-extractor:
+    runs-on: ubuntu-24.04
+    defaults:
+      run:
+        working-directory: src/cloud-providers
+    steps:
+      - name: Checkout the pull request code
+        uses: actions/checkout@1af3b93b6815bc44a9784bd300feb67ff0d1eeb3 # v6.0.0
+        with:
+          persist-credentials: false
+
+      - name: Read properties from versions.yaml
+        run: |
+          go_version="$(yq '.tools.golang' ../cloud-api-adaptor/versions.yaml)"
+          [ -n "$go_version" ]
+          echo "GO_VERSION=${go_version}" >> "$GITHUB_ENV"
+
+      - name: Setup Golang version ${{ env.GO_VERSION }}
+        uses: actions/setup-go@44694675825211faa026b3c33043df3e48a5fa00 # v6.0.0
+        with:
+          go-version: ${{ env.GO_VERSION }}
+          cache-dependency-path: "**/go.sum"
+
+      - name: Build config-extractor
+        run: make config-extractor
+
+      - name: Test all providers parse successfully
+        run: |
+          # Find all directories with manager.go (excluding cmd and util)
+          for provider_dir in */; do
+            provider=$(basename "$provider_dir")
+
+            # Skip non-provider directories
+            if [[ "$provider" == "cmd" || "$provider" == "util" ]]; then
+              continue
+            fi
+
+            # Check if manager.go exists
+            if [[ ! -f "${provider}/manager.go" ]]; then
+              continue
```

Comment:
Would it be worth error-ing here? It feels like if the manager.go file is missing then either we've refactored and need to add it to the non-provider directory, or a half-baked cloud-provider has gone it which is error worthy? Or am I missing something?

Reply:
I think its ok to bail here. Doing ....